datadog/resource_datadog_integration_slack_channel.go

```diff
@@ -0,0 +1,200 @@
+}
+
+func resourceDatadogIntegrationSlackChannelRead(d *schema.ResourceData, meta interface{}) error {
+	providerConf := meta.(*ProviderConfiguration)
+	datadogClient := providerConf.DatadogClientV1
+	auth := providerConf.AuthV1
+
+	teamName, channelName, err := teamNameAndChannelNameFromID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	slackChannel, httpResp, err := datadogClient.SlackIntegrationApi.GetSlackIntegrationChannel(auth, teamName, channelName).Execute()
+	if err != nil {
+		if httpResp.StatusCode == 404 {
+			d.SetId("")
+			return nil
+		}
+		return utils.TranslateClientError(err, "error getting slack channel")
+	}
+
+	return updateSlackChannelState(d, &slackChannel)
+}
+
+func resourceDatadogIntegrationSlackChannelUpdate(d *schema.ResourceData, meta interface{}) error {
+	providerConf := meta.(*ProviderConfiguration)
+	datadogClient := providerConf.DatadogClientV1
+	auth := providerConf.AuthV1
+
+	ddObject, err := buildDatadogSlackChannel(d)
+	teamName, channelName, err := teamNameAndChannelNameFromID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	slackChannel, _, err := datadogClient.SlackIntegrationApi.UpdateSlackIntegrationChannel(auth, teamName, channelName).Body(*ddObject).Execute()
+	if err != nil {
+		return utils.TranslateClientError(err, "error updating slack channel")
+	}
+
+	// Handle case where channel name is updated
+	d.SetId(fmt.Sprintf("%s:%s", teamName, slackChannel.GetName()))
+
+	return updateSlackChannelState(d, &slackChannel)
+}
+
+func resourceDatadogIntegrationSlackChannelDelete(d *schema.ResourceData, meta interface{}) error {
+	providerConf := meta.(*ProviderConfiguration)
+	datadogClient := providerConf.DatadogClientV1
+	auth := providerConf.AuthV1
+
+	teamName, channelName, err := teamNameAndChannelNameFromID(d.Id())
+
+	_, err = datadogClient.SlackIntegrationApi.RemoveSlackIntegrationChannel(auth, teamName, channelName).Execute()
+
+	if err != nil {
+		return utils.TranslateClientError(err, "error deleting slack channel")
+	}
+
+	return nil
+}
+
+func updateSlackChannelState(d *schema.ResourceData, slackChannel *datadogV1.SlackIntegrationChannel) error {
+	if err := d.Set("channel_name", slackChannel.GetName()); err != nil {
+		return err
+	}
+
+	tfChannelDisplay := buildTerraformSlackChannelDisplay(slackChannel.GetDisplay())
+	if err := d.Set("display", []map[string]interface{}{tfChannelDisplay}); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func buildTerraformSlackChannelDisplay(ddChannelDisplay datadogV1.SlackIntegrationChannelDisplay) map[string]interface{} {
+	tfChannelDisplay := map[string]interface{}{}
+	tfChannelDisplay["message"] = ddChannelDisplay.GetMessage()
+	tfChannelDisplay["notified"] = ddChannelDisplay.GetNotified()
+	tfChannelDisplay["snapshot"] = ddChannelDisplay.GetSnapshot()
+	tfChannelDisplay["tags"] = ddChannelDisplay.GetTags()
+
+	return tfChannelDisplay
+}
+
+func teamNameAndChannelNameFromID(id string) (string, string, error) {
```

Comment:
Would be good to add a unit test for this and maybe move to the internal/utils.go and internal/utils_test.go files?